### PR TITLE
 kql tweaks

### PR DESCRIPTION
**Pull request by AsafMah**
### Added
* Mgmt now supports kql.Statement
* kql.FromBuilder to deplicate builders
### Changed
* Moved all examples and code to use the the new kql.Statement (except for code explictly testing/showing the old Stmt)
* Moved all examples away from deprecated methods
* kql.NewBuilder renamed simply to kql.New
### Fixed
* Made AddUnsafe public
### Removed
* kusto.NewStmt is now deprecated
### Security

### CODE REVIEW

**Review by ronmonetaMicro — APPROVED** [review]

File: kusto/doc.go [hunk 1]
```diff
@@ -11,216 +11,328 @@ For details on the Azure Kusto service, see: https://azure.microsoft.com/en-us/s
...
+```
+
+### Ingestion
+
+The `ingest` package provides access to Kusto's ingestion service for importing data into Kusto. This requires
+some prerequisite knowledge of acceptable data formats, mapping references, etc.
+
+That documentation can be found [here](https://docs.microsoft.com/en-us/azure/kusto/management/data-ingestion/)
+
+Kusto's ingestion service makes no guarantees on when the data will show up in the table and is optimized for
```
Comment (yogilad) [inline kusto/doc.go:247]: Ha?

File: kusto/kusto.go [hunk 2]
```diff
@@ -294,10 +295,14 @@ func setQueryOptions(ctx context.Context, op errors.Op, query Statement, options
 	return opt, nil
 }
 
-func setMgmtOptions(ctx context.Context, op errors.Op, query Stmt, options ...MgmtOption) (*mgmtOptions, error) {
-	params, err := query.params.toParameters(query.defs)
-	if err != nil {
-		return nil, errors.ES(op, errors.KClientArgs, "QueryValues in the the Stmt were incorrect: %s", err).SetNoRetry()
+func setMgmtOptions(ctx context.Context, op errors.Op, query Statement, options ...MgmtOption) (*mgmtOptions, error) {
+	var params map[string]string
+	if stmt, ok := query.(Stmt); ok {
```
Comment (yogilad) [inline kusto/kusto.go:300]: Statements support options but not parameters.
Not sure I'm following what's going on here

File: kusto/doc.go [hunk 3]
```diff
@@ -11,216 +11,328 @@ For details on the Azure Kusto service, see: https://azure.microsoft.com/en-us/s
...
-	)
 	if err != nil {
-		panic("add error handling")
+	    panic("add error handling")
 	}
 
-# Querying with StatementBuilder
-Another way to build your query is using the StatementBuilder struct. With this struct you build your query as a combination of literals and special entities.
-This is mostly applicable when your KQL commands which do not support query parameters, or for situations where Kusto entities like database, table, column or function need to be safely added from variables.
+// You can see the generated parameters using the ToDeclarationString() method:
```
[the remark this answers is not in the document]
Comment (AsafMah) [reply kusto/doc.go:123]: That's basically the users way to see what parameters they have put in + it shows what happens behind the scenes

File: README.md [hunk 4]
```diff
@@ -336,37 +345,6 @@ if err := in.Stream(ctx, jsonEncodedData, ingest.JSON, "mappingName"); err != ni
 }
 ```
 
-#### Addtional Ingest Clients
```
[the remark this answers is not in the document]
Comment (AsafMah) [reply README.md:339]: These are release notes that accidentally remained in the file. I'll add it to the other section

File: kusto/kusto.go (hunk 2 above)
Comment (AsafMah) [reply kusto/kusto.go:299]: Fixed

**Pull request merged**